libafl/src/executors/afl_args.rs

```diff
@@ -0,0 +1,220 @@
+//! Share implementation of afl style arguments
+
+use alloc::{borrow::ToOwned, vec::Vec};
+use std::{
+    ffi::{OsStr, OsString},
+    path::Path,
+};
+
+use libafl_bolts::fs::{InputFile, get_unique_std_input_file};
+
+/// How to deliver input to an external program
+/// `StdIn`: The target reads from stdin
+/// `File`: The target reads from the specified [`InputFile`]
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub enum InputLocation {
+    /// Mutate a commandline argument to deliver an input
+    Arg {
+        /// The offset of the argument to mutate
+        argnum: usize,
+    },
+    /// Deliver input via `StdIn`
+    #[default]
+    StdIn,
+    /// Deliver the input via the specified [`InputFile`]
+    /// You can use specify [`InputFile::create(INPUTFILE_STD)`] to use a default filename.
+    File {
+        /// The file to write input to. The target should read input from this location.
+        out_file: InputFile,
+    },
+}
+
+/// The main implementation trait of afl style arguments handling
+pub trait AflTargetArgs: Sized {
+    /// Gets the arguments
+    fn arguments_ref(&self) -> &Vec<OsString>;
+    /// Gets the mutable arguments
+    fn arguments_mut(&mut self) -> &mut Vec<OsString>;
+
+    /// Gets the main program
+    fn program_ref(&self) -> &Option<OsString>;
+    /// Gets the mutable main program
+    fn program_mut(&mut self) -> &mut Option<OsString>;
+
+    /// Gets the input file
+    fn input_location_ref(&self) -> &InputLocation;
+    /// Gets the mutable input file
+    fn input_location_mut(&mut self) -> &mut InputLocation;
+
+    /// Get the environments
+    fn envs_ref(&self) -> &Vec<(OsString, OsString)>;
+    /// Get the mutable environments
+    fn envs_mut(&mut self) -> &mut Vec<(OsString, OsString)>;
+
+    /// Adds an environmental var to the harness's commandline
+    #[must_use]
+    fn env<K, V>(mut self, key: K, val: V) -> Self
+    where
+        K: AsRef<OsStr>,
+        V: AsRef<OsStr>,
+    {
+        self.envs_mut()
```

Comment:
Should this have special handling for AFL_PRELOAD?

Reply:
From what I understand, 1) it is handled by users 2) it doesn't need special handling (?) since it is just a common AFL env supported by fsrv?